lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1529,14 +1548,20 @@ class NGPHistoryParser {
         continue;
       }
 
+      // We need the timestamp of the time the Bolus event was programmed, not
+      // when it finished. Even for a normal basal, delivery for 100g of carbs can
+      // take a few minutes.
+      const programmedBolus = this.findMatchingBolusProgrammedEvent(event);
+      const beginTimestamp = _.isUndefined(programmedBolus) ? event.timestamp :
```

Comment:
Maybe we can factor all of these out into a `getBeginTimestamp` function? 

How rare would it be to have an bolus without a programmed event, where we only have the event timestamp? Should we maybe add an annotation, payload note or just a debug statement when it can't find the programmed timestamp?

Reply:
Yes, good idea about a single function.

It would be unusual, but possible. In fact, maybe we should completely discard these events in this case (similar to the suspends without matching suppressed basal info).